File: src/main/java/pl/dmichalski/algorithms/_1_sum_from_1_to_n/SumCounterService2.java
package pl.dmichalski.algorithms._1_sum_from_1_to_n;

class SumCounterService2 {

    /**
     * O(1) complexity
     */
    int countSumFrom1ToN(int n) {
        return n * (n + 1) / 2;
    }

}

File: src/main/kotlin/pl/dmichalski/algorithms/algorithms/_1_sum_from_1_to_n/SumCounterService2.kt
package pl.dmichalski.algorithms.algorithms._1_sum_from_1_to_n

internal class SumCounterService2 {

    /**
     * O(1) time complexity
     */
    fun countSumFrom1ToN(n: Int): Int {
        return n * (n + 1) / 2
    }
}
